### DISCUSSION TITLE
How do we trigger a service worker update prompt?

### DISCUSSION THREAD

**Discussion by tony19** [serwist/serwist#63] [Q&A]
Given the [service worker update subscription](https://github.com/serwist/serwist/discussions/62#discussion-6165140), how do we test that this component actually works properly? Do you have a sample app that demonstrates this?

I tried the following unsuccessfully:

1. Build app with the component mentioned above, using `window.serwist` instead of `window.workbox`. Make sure the app is in the root component's layout.
2. Serve app with `npm start` (runs `next start`)
3. Open browser to app and navigate to a few pages to activate service worker and cache some pages
4. Edit a client-only page (e.g., add a `console.log` to the home page component).
5. Rebuild app.
6. Restart app server with `npm start` (runs `next start`)
7. Navigate to different page in app (same browser tab from step 3)
8. Verify prompt appears to refresh app to install new service worker update. ❌

**Comment by DuCanhGH** [comment]
@tony19 doing it manually as described is how I would usually do it. There are probably alternative ways, but I'm in the dark about that. Now, if that example didn't work for you, check if you did set `@serwist/next.withSerwistInit.PluginOptions.register` and `@serwist/sw.installSerwist.SerwistOptions.skipWaiting` to `false`.

**Reply by tony19** [reply]
Those settings were the key. Thanks!
